### ISSUE TITLE
useless_format: .to_string() suggested twice: PathBuf.display().to_string()

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang-nursery/rust-clippy#3020]
````rust
use std::path::PathBuf;

fn main() {
    let x = PathBuf::from("/bar/foo/qux");
    let y = format!("{}", x.display().to_string());
    println!("{}", y);
}
````
causes
````
warning: useless use of `format!`
 --> src/main.rs:6:13
  |
6 |     let y = format!("{}", x.display().to_string());
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using .to_string(): `x.display().to_string().to_string()`
  |
  = note: #[warn(useless_format)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#useless_format
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
    Finished dev [unoptimized + debuginfo] target(s) in 0.12s
````
which looks a bit goofy

**Comment by oli-obk** [comment]
We're currently just checking the type of the formatting argument. Maybe we should just have another lint that complains about any formatting argument that ends in a `to_string()` call?
